### PR TITLE
fix: multiple issue fixes

### LINKED ISSUE #24: Import not generated
[`ObjectErrors`](https://github.com/fireflysemantics/validator/blob/master/src/container/error/ObjectErrors.ts) import `ValidationError`.

When the projects build is run, the import is not generated.  I think it's because `ObjectErrors` and `ValidationError` both reside in the same directory.

If I do not use a `paths` import, but instead switch it to a relative import, then the build generates the import.

To see the current behavior:
```
git clone git@github.com:fireflysemantics/validator.git
npm run dist
```

Look at `dist/container/error/ObjectErrors.d.ts`.  It will be missing the `ValidationError` import.

### LINKED ISSUE #48: Importing a mix of types and values in an import using a custom path results in undefined imports in result JavaScript
Given:

file1.ts:
```
export type Foo = 'bar';

export const baz = 'inga'; // ;)
```

file2.ts:
```
import { Foo, baz } from '@custom/scope/file1';

[...]
```

When the code is compiled, the Javascript import will read something like this:

file2.js:
```
import { Foo, baz } from '@custom/scope/file1';
```

but `Foo` will be undefined and cause errors.

### LINKED ISSUE #60: Only transforms if relative to first path in the array
Example

`./index.ts`
```ts
import { v1 } from "#path/b"  // Resolves to ./a.ts
import { v2 } from "#path/a"  // Resolves to ./dir2/b.ts

export { v1, v2 }
```

`./tsconfig.json`
```jsonc
{
  "compilerOptions": {
    // ...
    "baseUrl": ".",
    "paths": {
      "#path/*": [ "./*", "./dir2/*" ]
    },
    "plugins": [
      {
        "transform": "typescript-transform-paths"
      }
    ]
  }
}
```

Output: `index.js` (I used ESNext for readability, but happens with CommonJS as well)
```ts
import { gg } from "#path/b";  // Does not transform because it's the second path in the array
import { ga } from "./a";
export { gg, ga };
```

### PR DESCRIPTION
**Pull request by nonara**
- Allow more than one path routing (fixes #60)
- Remove implicit extensions from output (fixes #24)
- Properly implemented isTypeOnly (fixes #48)
- Corrected errors in tests due to TS changing logic for type only star exports
- Bonus: Made package zero-dependency

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -96,17 +116,6 @@ const transformer = (_: ts.Program) => (context: ts.TransformationContext) => (
     node.arguments.length === 1;
 
   function visit(node: ts.Node): ts.VisitResult<ts.Node> {
```
Comment (nonara) [inline src/index.ts:118]: ## Note on TS changes regarding type only exports

It seems that types only export behaviour has changed in TypeScript. For one, they added `isTypeOnly`, but they've also changed the standard behaviour of `export * from 'module'` and `import 'module'`

Even if `module` has only type exports, this line will still get compiled into the final result. This was causing the tests to fail, by default, after cloning the package. In order to match the prescribed TS behaviour , we've added the `isTypeOnly` param where needed as well as removed the following conditional.

In short, this means the TS compiler has taken over this logic.

File: tests/__fixtures/with-path/core/index.ts [hunk 2]
```diff
@@ -1,5 +1,7 @@
 import sum = require("@utils/sum");
 export { sum } from "@utils";
```
Comment (nonara) [inline tests/__fixtures/with-path/core/index.ts:2]: Allows testing for secondary path routing

**Comment by danielpza** [comment]
That's great @nonara , I'll definitely take my time in the evening to proper review it and test it.

**Comment by danielpza** [comment]
@all-contributors add @nonara for bug, test, and code

**Comment by nonara** [comment]
Thanks. 😄

**Comment by nonara** [comment]
Whoops had the wrong issue number on one. Tagged an already closed issue

Path extensions also fixes #59

**Comment by danielpza** [comment]
The fix is up in v1.1.15 :tada:, many thanks for the amazing contribution @nonara!
